File: app/src/main/java/crabster/rudakov/sberschoollesson30hwk/domain/converter/IDictionaryConverter.java
package crabster.rudakov.sberschoollesson30hwk.domain.converter;

import java.util.*;

import crabster.rudakov.sberschoollesson30hwk.domain.models.*;
import crabster.rudakov.sberschoollesson30hwk.domain.repositories.models.*;

public interface IDictionaryConverter {

    DictionaryItemModel convert(DictionaryItem dictionaryItem);

    DictionaryItem reverse(DictionaryItemModel dictionaryItemModel);

    List<DictionaryItem> reverseList(List<DictionaryItemModel> dictionaryItemModels);

}

File: app/src/main/java/crabster/rudakov/sberschoollesson30hwk/domain/converter/IDictionaryConverter.kt
package crabster.rudakov.sberschoollesson30hwk.domain.converter

import crabster.rudakov.sberschoollesson30hwk.domain.models.DictionaryItem
import crabster.rudakov.sberschoollesson30hwk.domain.repositories.models.DictionaryItemModel


/**
 * Converter interface for model classes [DictionaryItem] and [DictionaryItemModel]
 */
interface IDictionaryConverter {

    fun convert(dictionaryItem: DictionaryItem): DictionaryItemModel

    fun reverse(dictionaryItemModel: DictionaryItemModel): DictionaryItem

    fun reverseList(dictionaryItemModels: List<DictionaryItemModel>): List<DictionaryItem>

}